src/main/routing/topology.c

```diff
@@ -241,6 +243,30 @@ static gboolean _topology_isValidEdgeAttributeKey(const gchar* attrName, EdgeAtt
     return (r == 0) ? TRUE : FALSE;
 }
 
+static gboolean _topology_findVertexAttributeStringBandwidth(Topology* top, igraph_integer_t vertexIndex,
+        VertexAttribute attr, guint64* valueOut) {
+    MAGIC_ASSERT(top);
+
+    const gchar* name = _topology_vertexAttributeToString(attr);
+
+    if (igraph_cattribute_has_attr(&top->graph, IGRAPH_ATTRIBUTE_VERTEX, name)) {
+        const gchar* value = igraph_cattribute_VAS(&top->graph, name, vertexIndex);
+        if (value != NULL && value[0] != '\0') {
+            if (valueOut != NULL) {
+                int64_t bandwidth = parse_bandwidth(value);
+                if (bandwidth >= 0) {
+                    // parse_bandwidth() returns bits-per-second, but shadow works with KiB/s
+                    bandwidth /= 8 * 1024;
+                    *valueOut = bandwidth;
+                    return TRUE;
```

Comment:
I think we are planning to remove KiB/s and just pass around bits or bytes. Do we need a TODO here for when we address that?

Reply:
Added a TODO (and in `_controller_registerHostCallback()` also).